src/device-state/current-state.ts

```diff
@@ -203,14 +203,10 @@ const reportCurrentState = (): null => {
 				...currentDeviceState.dependent,
 			};
 
-			const stateDiff = getStateDiff();
-			if (_.size(stateDiff) === 0) {
-				reportPending = false;
-				return null;
-			}
-
+			// Report current state
 			await report();
-			reportCurrentState();
```

Comment:
So before this, the function would fall into this infinite loop and the interval call on the `startReporting` would always fail. Weird we never caught it before.

Reply:
I had a conversation with Page about this. The system made sense before because changes were pushed rather then queried but with the addition of device metrics we now had a need to query for metrics on an interval. That's why the startReporting function is odd to have onChange and on interval.

https://github.com/balena-os/balena-supervisor/commit/eeda7e3ab17dc68d592acc2e4f58f20dad9a677a#diff-4201a9052aedc0dfdf3d8eb0566ecc3ac822602e1a872e858cddecb2ac772b8cR204-R214